content/reference/sql/version-control/dolt-sql-functions.md

```diff
@@ -656,3 +657,159 @@ returns just `D`. `DOLT_LOG('main...feature')` and `DOLT_LOG('feature...main')`
 both return `F`, `E`, and `D`.
 
 Learn more about two vs three dot log [here](https://www.dolthub.com/blog/2022-11-11-two-and-three-dot-diff-and-log).
+
+## `DOLT_PATCH()`
+
+Generate the SQL statements needed to patch a table (or all tables) from a starting revision 
+to a target revision. This can be useful when you want to import data into Dolt from an external source, 
+compare differences, and generate the SQL statements needed to patch the original source. This command is
+equivalent of [`dolt diff -r sql` CLI command](../../cli.md#dolt-diff).
+Both schema and/or data diff statements are returned if applicable. Some data diff cannot be
+produced from incompatible schema changes; these are shown as warnings containing
+which table this occurred on.
+
+The order of the statements is that the schema patch comes first after the data patch. If patching all tables,
+then we recommend to turn off the foreign key checks before applying these patch statements in order of
+them returned to avoid conflicts.
+
+Getting SQL patch statements is only available as table function for now;
+the CLI `dolt patch` command will be supported in the future.
+
+### Privileges
+
+`DOLT_PATCH()` table function requires `SELECT` privilege for all tables if no table is defined or
+for the defined table only.
+
+### Options
+
+```sql
+DOLT_PATCH(<from_revision>, <to_revision>, <optional_tablename>)
+DOLT_PATCH(<from_revision..to_revision>, <optional_tablename>)
+DOLT_PATCH(<from_revision...to_revision>, <optional_tablename>)
+```
+
+The `DOLT_PATCH()` table function takes three arguments:
```

Comment:
Maybe just something like below, since it doesn't always take three arguments? 
```suggestion
The `DOLT_PATCH()` table function takes the following arguments:
```